src/cryptography/hazmat/primitives/asymmetric/dh.py

```diff
@@ -87,9 +87,6 @@ def __init__(self, p, g, q=None):
         if q is not None and not isinstance(q, six.integer_types):
             raise TypeError("q must be integer or None")
 
-        if q is None and g not in (2, 5):
-            raise ValueError("DH generator must be 2 or 5")
```

Comment:
Indeed we do not use DHParameterNumbers unless for immediate conversion to DHParameters. We'd be fine with the workaround you suggest. Do you see anything else to add to the PR?